### PR TITLE
Remodel `EventLoop` as a future instead of as a struct

### PR DESCRIPTION
**Pull request by thomaseizinger**
TODO: Commits need re-ordering / squashing before this can go in. Too many things changed with the latest rebase. Converted to draft for now.

After some discussions in the libp2p repo [here](https://github.com/libp2p/rust-libp2p/discussions/2024), I thought this design might actually be easier to understand because it involves less state in a way by moving things to local variables.

The naming is also slightly more accurate IMO because the variables `event_loop` now actually contains a `Future` that contains a loop! Before it was a struct that you had to call `run` on.

### CODE REVIEW

**Review by rishflab — APPROVED** [review]

File: swap/src/protocol/alice/event_loop.rs [hunk 1]
```diff
@@ -105,34 +106,10 @@ pub async fn new<LR>(
...
-                        }
-
-                        let state0 = match State0::new(btc, xmr, env_config, bitcoin_wallet.as_ref(), &mut OsRng).await {
-                            Ok(state) => state,
-                            Err(e) => {
-                                tracing::warn!(%peer, "Failed to make State0 for execution setup: {:#}", e);
-                                continue;
-                            }
+                    SwarmEvent::Behaviour(OutEvent::SpotPriceRequested { request: spot_price::Request { btc }, channel, peer }) => {
+                        if let Err(e) = handle_spot_price_request(&mut latest_rate, &mut swarm, monero_wallet.as_ref(), bitcoin_wallet.as_ref(), env_config, channel, peer, btc, max_buy).await {
```
Comment (da-kami) [inline swap/src/protocol/alice/event_loop.rs:110]: Looks like `fmt` did not pick up the formatting in the `select!` - might be worth manually adding a breakline.

**Review by da-kami — APPROVED** [review]

**Comment by thomaseizinger** [comment]
@da-kami @rishflab force pushed to resolve conflicts but also added 0c47cb1 on top of it because I wasn't quite happy with the loss of documentation due to the removal of fields. It is quite similar to what it was before :sweat_smile: 

I think it is still better though :)

File: swap/src/protocol/alice/event_loop.rs (hunk 1 above)
Comment (thomaseizinger) [reply swap/src/protocol/alice/event_loop.rs:101]: Yeah, anything in the macro is never formatted. because it is opaque code to the formatter.

**Review by da-kami — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Going to close this for now due to being stale and maybe revisit another time.

**Pull request closed without merging**